endpoint/endpoint.go

```diff
@@ -107,6 +107,9 @@ func (t Targets) IsLess(o Targets) bool {
 	return false
 }
 
+// ProviderSpecific holds configuration which is specific to individual DNS providers
+type ProviderSpecific map[string]string
```

Comment:
Please reconsider using list of k-v values as per kubernetes api conventions. please refer to this comment for more information https://github.com/kubernetes-incubator/external-dns/pull/657#discussion_r211792059

Reply:
I'll take a look when I'm back from holidays (end of September)

Reply:
I've updated the type and updated the CRD docs. Could you take a look specifically if the changes to these docs are correct?